p2panda-rs/src/schema/schema.rs

```diff
@@ -60,6 +63,46 @@ impl Schema {
     pub fn as_cddl(&self) -> String {
         generate_cddl_definition(&self.fields)
     }
+
+    /// Access the schema's [`SchemaId`].
+    #[allow(unused)]
+    pub fn id(&self) -> &SchemaId {
+        &self.id
+    }
+
+    /// Access the schema view id.
+    #[allow(unused)]
+    pub fn view_id(&self) -> &DocumentViewId {
+        if let SchemaId::Application(_, view_id) = &self.id {
+            return view_id;
+        }
+        // This statement is unreachable because it's not possible to construct a `Schema`
+        // that doesn't contain an application schema id.
+        panic!()
```

Comment:
I can see how this is hard to avoid in this case. Hmn. 🤔

How about keeping the name and document view in the struct, returning it normally here and constructing a schema id when calling `id` with these two values?

You could even keep schema id in the struct as well, I know its maybe redundant but less hacky?

Reply:
That's possible of course. I don't like being that redundant :(. But I think here that's the best option, I can change it.

It's a similar situation [here](https://github.com/p2panda/p2panda/blob/c6d76afe789dbbad5cd71ace57de6f615489ef9b/p2panda-rs/src/permissions/key_group/views/key_group.rs#L47-L54) and I really like making system schema views a thin wrapper of a document view and not replicating all the data.

Reply:
Or you just unwrap and give a note why you do it? (You can unwrap `Option`)

Reply:
Hmmm, yeh, I see the problem, that's annoying. I also really like system views being very thin wrappers, that was the original idea... but I would probably prefer to not have unreachable code like this :-( 

Could we have a `view_id()` method on `SchemaId` which returns a view id wrapped in an option? Then we just unwrap here as we know it will be some in this case?

Reply:
(didn't see your comment adz, that works too)